### ISSUE TITLE
HMR not working

### ISSUE THREAD

**Issue by mp3por** [crxjs/chrome-extension-tools#892]
### Build tool

Rollup

### Where do you see the problem?

- [X] In the browser
- [ ] In the terminal

### Describe the bug

When I open the PopUp and then make a change in the code base, the pop-up does not reflect that change until I manually refresh the popup by closing and opening it again.

### Reproduction

https://github.com/JohnBra/vite-web-extension

### System Info

```shell
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Max
    Memory: 102.91 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.12.2 - ~/.nvm/versions/node/v20.12.2/bin/node
    npm: 10.5.0 - ~/.nvm/versions/node/v20.12.2/bin/npm
    pnpm: 9.1.2 - /opt/homebrew/bin/pnpm
    Watchman: 2024.05.06.00 - /opt/homebrew/bin/watchman
  Browsers:
    Brave Browser: 125.1.66.118
    Chrome: 125.0.6422.142
    Safari: 17.5
  npmPackages:
    @crxjs/vite-plugin: ^2.0.0-beta.23 => 2.0.0-beta.23
    vite: ^5.2.13 => 5.2.13
```

### Severity

annoyance

**Comment by Toumash** [comment]
Im not sure why i cannot run your extension on windows, but you're clearly **not using** vite for HMR. 
Nodemon works by watching files changing and rebuilding the whole thing without hmr.

![image](https://github.com/crxjs/chrome-extension-tools/assets/9840635/beceb723-e194-49b9-b747-ac14d307b50d)

To use any vite dev feature (including hmr) use vite dev mode.
![image](https://github.com/crxjs/chrome-extension-tools/assets/9840635/adb467d5-28bb-4307-8709-341b8343d061)

**Comment by Toumash** [comment]
I have no idea what does nowork with fs-extra in your project but i've managed it to work no problem. Adjust the vite.config to your needs - i've removed some things just to not debug it faster


https://github.com/JohnBra/vite-web-extension/pull/34
<details>
<summary> demo.gif</summary>
![demo](https://github.com/crxjs/chrome-extension-tools/assets/9840635/9c6ff069-dd8b-4a3e-ab7f-cb1926fb33bd)
</details>

**Comment by mp3por** [comment]
> I have no idea what does nowork with fs-extra in your project but i've managed it to work no problem. Adjust the vite.config to your needs - i've removed some things just to not debug it faster
> 
> If that works for you then lets close this issue @mp3por
> 
> [JohnBra/vite-web-extension#34](https://github.com/JohnBra/vite-web-extension/pull/34)
> 
> demo.gif

I will check during the weekend and come back to you.

**Comment by Toumash** [comment]
@mp3por and?

**Comment by mp3por** [comment]
@Toumash 

I was not able to get the HMR working. When I change my code to look like yours however I get the following issue. This screen keeps flickering. I don't understand enough to debug further. 

<img width="425" alt="image" src="https://github.com/crxjs/chrome-extension-tools/assets/5836859/477ed8e0-709b-4046-97af-c2c316327509">

**Comment by mtliendo** [comment]
Same issue here on mac m3 running node v22

I'm using vite 3 with ` "@crxjs/vite-plugin": "^2.0.0-beta.25",`.

my manifest is:
```json
{
	"manifest_version": 3,
	"name": "CRXJS React Vite Example",
	"version": "1.0.0",
	"action": { "default_popup": "index.html" }
}
```

According to the docs I should be able to run `npm run dev` and assign the generated `dist` folder to be my chrome extension. But doing so results in the flickering image described above.

If I run `npm run build`, the correct contents will be generated in the extension, but that of course doesn't use HMR.

In short, when running `npm run dev` using vite3, the contents are never loaded and instead the flickering occurs because the default loading-screen is never able to fetch the app contents.

**Comment by Toumash** [comment]
Thats something different #900
